packages/wonder-blocks-tabs/src/components/tab-panel.tsx

```diff
@@ -13,20 +16,55 @@ type Props = {
      * The id of the associated element with role="tab".
      */
     "aria-labelledby": string;
+
+    /**
+     * Whether the tab panel is active.
+     */
+    active?: boolean;
 };
 
+const StyledDiv = addStyle("div");
 /**
  * A component that has `role="tabpanel"` and is used to represent a tab panel
  * in a tabbed interface.
  */
-export const TabPanel = React.forwardRef(function TabPanel(
-    props: Props,
-    ref: React.ForwardedRef<HTMLDivElement>,
-) {
-    const {children, id, "aria-labelledby": ariaLabelledby} = props;
+export const TabPanel = (props: Props) => {
+    const {
+        children,
+        id,
+        "aria-labelledby": ariaLabelledby,
+        active = false,
+    } = props;
+
+    const ref = React.useRef<HTMLDivElement>(null);
+    const [hasFocusableElement, setHasFocusableElement] = React.useState(false);
+
+    React.useEffect(() => {
+        // Whenever tab panel contents change, determine if the panel has a
+        // focusable element
+        if (ref.current) {
+            setHasFocusableElement(findFocusableNodes(ref.current).length > 0);
+        }
+    }, [active, ref, children]);
+
     return (
-        <div ref={ref} role="tabpanel" id={id} aria-labelledby={ariaLabelledby}>
+        <StyledDiv
+            ref={ref}
+            role="tabpanel"
+            id={id}
+            aria-labelledby={ariaLabelledby}
+            style={!active && styles.hidden}
+            // If the tab panel doesn't have focusable elements, it should be
+            // focusable so that it is included in the tab sequence of the page
+            tabIndex={hasFocusableElement ? undefined : 0}
+        >
             {children}
-        </div>
+        </StyledDiv>
     );
+};
+
+const styles = StyleSheet.create({
+    hidden: {
```

Comment:
This would be a good candidate for `wonder-blocks-styles`! It could also be defined with the HTML `hidden` attribute (e.g. `<div hidden>`).

Reply:
Updated to use the `hidden` attribute instead (TIL!). I also made a note in https://khanacademy.atlassian.net/browse/WB-1776 that it'd be helpful to also have global `display: none` styles